x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -402,8 +405,18 @@ public void executePlan(
                             exchangeSinkSupplier
                         ),
                         coordinatorPlan,
-                        localListener.acquireCompute()
+                        delegate.delegateFailure((l, r) -> {
+                            l.onResponse(r);
+                            if (configuration.profile() == false && exchangeSource.isFinished()) {
+                                threadPool.scheduleUnlessShuttingDown(
```

Comment:
Could you add a comment here that this is to discard outstanding responses after the timeout?

Reply:
Just checking - this listener is called for all responses and checking for the exchangeSource being finished makes sure this is the response from the coordinating driver. Would it be less racy to check that the response contains the coordinating node driver? Maybe we can a `boolean` to `DriverCompletionInfo` that is true if it contains the coordinating node driver. Or is this safe in a way I don't understand?

Reply:
github diff view does not show the entire call importunately:
https://github.com/elastic/elasticsearch/blob/54ef16568a10f9bbe65a3bfd38048a6ba0b70a35/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java#L392-L406

This entire call is only happening for the final reduction on coordinating node. We do not perform this check on any other driver.

Reply:
:+1: